tt_metal/api/device.hpp

```diff
@@ -0,0 +1,342 @@
+
+/**
+ * @brief Computes the storage grid size for the device.
+ *
+ * @param device The device to query.
+ * @return CoreCoord representing the storage grid size.
+ */
+CoreCoord GetComputeWithStorageGridSize(Device device);
+
+/**
+ * @brief Retrieves the DRAM grid size for the device.
+ *
+ * @param device The device to query.
+ * @return CoreCoord representing the DRAM grid size.
+ */
+CoreCoord GetDramGridSize(Device device);
+
+/**
+ * @brief Converts a logical core coordinate to a physical core coordinate.
+ *
+ * @param device The device to query.
+ * @param logical_core The logical core coordinate.
+ * @param core_type The type of the core.
+ * @return CoreCoord representing the physical core coordinate.
+ */
+CoreCoord PhysicalCoreFromLogical(Device device, const CoreCoord &logical_core, const CoreType &core_type);
+
+/**
+ * @brief Retrieves the worker core coordinate corresponding to a logical core.
+ *
+ * @param device The device to query.
+ * @param logical_core The logical core coordinate.
+ * @return CoreCoord representing the worker core coordinate.
+ */
+CoreCoord WorkerCoreFromLogical(Device device, const CoreCoord &logical_core);
+
+/**
+ * @brief Retrieves the Ethernet core coordinate corresponding to a logical core.
+ *
+ * @param device The device to query.
+ * @param logical_core The logical core coordinate.
+ * @return CoreCoord representing the Ethernet core coordinate.
+ */
+CoreCoord EthernetCoreFromLogical(Device device, const CoreCoord &logical_core);
+
+/**
+ * @brief Retrieves a list of Ethernet socket coordinates connected to a specific chip ID.
+ *
+ * @param device The device to query.
+ * @param connected_chip_id The connected chip ID.
+ * @return Vector of CoreCoord representing Ethernet socket coordinates.
+ */
+std::vector<CoreCoord> GetEthernetSockets(Device device, chip_id_t connected_chip_id);
+
+/**
+ * @brief Returns the number of banks for a specific buffer type on the device.
+ *
+ * @param device The device to query.
+ * @param buffer_type The type of buffer.
+ * @return Number of banks.
+ */
+uint32_t GetNumBanks(Device device, const BufferType &buffer_type);
+
+/**
+ * @brief Computes the offset of a specific bank for a buffer type on the device.
+ *
+ * @param device The device to query.
+ * @param buffer_type The type of buffer.
+ * @param bank_id The ID of the bank.
+ * @return Offset of the bank.
+ */
+int32_t GetBankOffset(Device device, BufferType buffer_type, uint32_t bank_id);
+
+/**
+ * @brief Retrieves bank IDs associated with a logical core for a given buffer type.
+ *
+ * @param device The device to query.
+ * @param buffer_type The type of buffer.
+ * @param logical_core The logical core coordinate.
+ * @return Reference to a vector of bank IDs.
+ */
+const std::vector<uint32_t> &BankIdsFromLogicalCore(Device device, BufferType buffer_type, const CoreCoord &logical_core);
+
+/**
+ * @brief Deallocates all buffers on the device.
+ */
+void DeallocateBuffers(Device device);
```

Comment:
move the gets below up w/ the other gets, this isn't a getter

Reply:
done